plugins/passwordagentauthplugin/passwordagentplugin.cpp

```diff
@@ -0,0 +1,235 @@
+
+Q_DECLARE_LOGGING_CATEGORY(lcPasswordAgent)
+Q_LOGGING_CATEGORY(lcPasswordAgent, "org.sailfishos.secrets.plugin.authentication.passwordagent", QtWarningMsg)
+
+static QDBusMessage passwordAgentMethodCall(const QString &member)
+{
+    return QDBusMessage::createMethodCall(
+                QStringLiteral("org.sailfishos.Lipstick.SecurityUi"),
+                QStringLiteral("/org/sailfishos/Lipstick/SecurityUi/PasswordAgent"),
+                QStringLiteral("org.sailfishos.Security.PasswordAgent"),
+                member);
+}
+
+class PasswordAgentResponse;
+class PasswordClientAdaptor : public QDBusAbstractAdaptor
+{
+    Q_OBJECT
+    Q_CLASSINFO("D-Bus Interface", "org.sailfishos.Security.PasswordClient")
+public:
+    explicit PasswordClientAdaptor(PasswordAgentResponse *response);
+
+public slots:
+    void VerifyPassword(const QString &password);
+
+private:
+    PasswordAgentResponse *m_response;
+};
+
+class PasswordAgentResponse : public QDBusPendingCallWatcher
+{
+public:
+    PasswordAgentResponse(
+            const QDBusObjectPath &object,
+            const QDBusPendingCall &call,
+            uint callerPid,
+            qint64 requestId,
+            const InteractionParameters &interactionParameters,
+            const QString &interactionServiceAddress,
+            QObject *parent)
+        : QDBusPendingCallWatcher(call, parent)
+        , interactionParameters(interactionParameters)
+        , serviceAddress(interactionServiceAddress)
+        , requestId(requestId)
+        , callerPid(callerPid)
+    {
+        QDBusConnection::systemBus().registerObject(object.path(), this);
+    }
+
+    void cancel()
+    {
+        QDBusConnection::systemBus().send(
+                    passwordAgentMethodCall(QStringLiteral("Cancel")) << QVariant::fromValue(object));
+    }
+
+    PasswordClientAdaptor clientAdaptor { this };
+    const QDBusObjectPath object;
+    const InteractionParameters interactionParameters;
+    const QString serviceAddress;
+    QByteArray password;
+    const qint64 requestId;
+    const uint callerPid;
+};
+
+PasswordClientAdaptor::PasswordClientAdaptor(PasswordAgentResponse *response)
+    : QDBusAbstractAdaptor(response)
+    , m_response(response)
+{
+}
+
+void PasswordClientAdaptor::VerifyPassword(const QString &password)
+{
+    m_response->password = password.toUtf8();
+}
+
+PasswordAgentPlugin::PasswordAgentPlugin(QObject *parent)
+    : AuthenticationPlugin(parent)
+{
+}
+
+PasswordAgentPlugin::~PasswordAgentPlugin()
+{
+    for (PasswordAgentResponse *response : m_responses) {
+        response->cancel();
+    }
+}
+
+AuthenticationPlugin::AuthenticationTypes
+PasswordAgentPlugin::authenticationTypes() const
+{
+    return AuthenticationPlugin::NoAuthentication;
+}
+
+InteractionParameters::InputTypes
+PasswordAgentPlugin::inputTypes() const
+{
+    return InteractionParameters::NumericInput
+            | InteractionParameters::AlphaNumericInput;
+}
+
+Result beginAuthentication(
```

Comment:
Missing the namespace here, should add ```PasswordAgentPlugin::```.

Reply:
Oops thanks, will fix